magefile.go

```diff
@@ -1001,7 +1001,19 @@ func (Cloud) Push() error {
 		tag = fmt.Sprintf("%s-%s-%d", version, commit, time)
 	}
 
+	fips := os.Getenv(fipsEnv)
+	defer os.Setenv(fipsEnv, fips)
+	fipsVal, err := strconv.ParseBool(fips)
+	if err != nil {
+		fipsVal = false
+	}
+	os.Setenv(fipsEnv, strconv.FormatBool(fipsVal))
+	devtools.FIPSBuild = fipsVal
```

Comment:
I think it would be helpful to add a comment here what this section is doing and why.

Reply:
I see you added a comment. Thanks.  Could you explain in there why you have the `defer os.Setenv(fipsEnv, fips)`?  That bit in particular wasn't obvious to me.

Reply:
It's copied from how we handle the env var in other parts of the magefile